protostar/cairo_testing/cairo_test_execution_state.py

```diff
@@ -10,45 +11,164 @@
+        address_to_class_hash: Optional[dict[Address, ClassHashType]] = None,
+        target_address_to_pranked_address: Optional[dict[Address, Address]] = None,
+    ) -> None:
+        self._emitted_events = emitted_events or []
+        self._class_hash_to_contract_abi = class_hash_to_contract_abi or {}
+        self._event_key_to_event_selector = event_key_to_event_selector or {}
+        self._event_selector_to_event_abi = event_selector_to_event_abi or {}
+        self._contract_address_to_class_hash = address_to_class_hash or {}
+        self._target_address_to_pranked_address = (
+            target_address_to_pranked_address or {}
+        )
+
+    def get_emitted_events(self) -> list[Event]:
+        return self._emitted_events
+
+    def get_pranked_address(self, target_address: Address) -> Optional[Address]:
+        if target_address in self._target_address_to_pranked_address:
+            return self._target_address_to_pranked_address[target_address]
+        return None
+
+    def get_class_hash_from_address(self, contract_address: Address) -> ClassHashType:
+        return self._contract_address_to_class_hash[contract_address]
+
+    def get_contract_abi_from_class_hash(self, class_hash: ClassHashType) -> AbiType:
+        return self._class_hash_to_contract_abi[class_hash]
+
+    def get_contract_abi_from_contract_address(
+        self, contract_address: Address
+    ) -> AbiType:
+        class_hash = self._contract_address_to_class_hash[contract_address]
+        return self._class_hash_to_contract_abi[class_hash]
+
+    def get_event_selector_from_event_key(self, key: int) -> Selector:
+        return self._event_key_to_event_selector[key]
+
+    def bind_event_key_to_event_selector(self, key: int, event_selector: Selector):
+        self._event_key_to_event_selector[key] = event_selector
+
+    def bind_contract_address_to_class_hash(
+        self, address: Address, class_hash: ClassHashType
+    ) -> None:
+        self._contract_address_to_class_hash[address] = class_hash
+
+    def bind_class_hash_to_contract_abi(
+        self, class_hash: ClassHashType, contract_abi: AbiType
+    ) -> None:
+        self._class_hash_to_contract_abi[class_hash] = contract_abi
+
+    def bind_event_selector_to_event_abi(
+        self, event_selector: Selector, event_abi: AbiType
+    ) -> None:
+        self._event_selector_to_event_abi[event_selector] = event_abi
+
+    def add_emitted_events(self, emitted_events: list[Event]) -> None:
+        self._emitted_events.extend(emitted_events)
+
+    def set_pranked_address(self, target_address: Address, pranked_address: Address):
+        self._target_address_to_pranked_address[target_address] = pranked_address
+
+    def remove_pranked_address(self, target_address: Address):
+        if target_address in self._target_address_to_pranked_address:
+            del self._target_address_to_pranked_address[target_address]
+
+    def clone(self):
```

Comment:
The naming is not consistent, we have `fork`, `copy`, and now `clone`. I don't see a particular difference between those.